File: app/models/treasury.py
from pydantic import BaseModel
from typing import Optional, List

class TreasuryForecastCreate(BaseModel):
    forecast_date: str
    estimated_inflow: float = 0
    estimated_outflow: float = 0
    notes: Optional[str] = ""

class BankStatementImportCreate(BaseModel):
    account_id: int
    filename: str

class CashFlowPoint(BaseModel):
    date: str
    inflow: float
    outflow: float
    net: float
    cumulative_net: float

File: frontend/src/app/api/treasury.ts
import { apiRequest } from "./client";

export type TreasuryProjectionRow = {
  date: string;
  inflow: number;
  outflow: number;
  net: number;
  cumulative_net: number;
};

export function getTreasuryProjection(token: string, days = 30) {
  return apiRequest<TreasuryProjectionRow[]>(`/treasury/projection?days=${encodeURIComponent(String(days))}`, { token });
}

export function createTreasuryForecast(
  token: string,
  payload: {
    forecast_date: string;
    estimated_inflow?: number;
    estimated_outflow?: number;
    notes?: string;
  }
) {
  return apiRequest<{ id: number; message: string }>("/treasury/forecasts", {
    method: "POST",
    token,
    body: payload
  });
}

export function logBankImport(
  token: string,
  payload: {
    account_id: number;
    filename: string;
  }
) {
  return apiRequest<{ id: number; message: string }>("/treasury/bank-imports", {
    method: "POST",
    token,
    body: payload
  });
}
